src/actions/classifier.js

```diff
@@ -55,120 +54,85 @@ export function startNewClassification(workflowID, firstClassification=true) {
           subjects: [subject.id]
         }
       })
-    }).then((classification) => {
-      dispatch(setState(`classifier.classification.${workflowID}`, classification))
-      dispatch(setState('classifier.isFetching', false))
-    }).catch(() => {
-      Alert.alert('Error', 'Sorry, but there was an error loading this workflow.  Please try again later.',
-        [{text: 'Go Back', onPress: () => { Actions.pop()}}]
-      )
     })
-  }
-}
-
-export function saveAnnotation(task, value) {
-  return (dispatch, getState) => {
-    const workflowID = getState().main.classifier.currentWorkflowID
-    dispatch(setState(`classifier.annotations.${workflowID}.${task}`, value))
-  }
+    .then((classification) => {
+      return dispatch(setClassificationForWorkflow(workflow.id, classification))
+    })
 }
 
-export function removeAnnotationValue(task, value) {
+export function saveThenStartNewClassification(workflow) {
   return (dispatch, getState) => {
-    const workflowID = getState().main.classifier.currentWorkflowID
-    dispatch(removeState(`classifier.annotations.${workflowID}.${task}`, value))
-  }
-}
+    const classifier = getState().classifier
+    const classification = classifier.classification[workflow.id]
+    const subject = classifier.subject[workflow.id]
+    const structureAnnotation = (a) => {
+      return {task: a[0], value: a[1] } 
+    }
+    const annotations = R.map(structureAnnotation, R.toPairs(classifier.annotations[workflow.id]))
```

Comment:
This seems very odd to me, yet I know it is the code that was previously implemented. Can you make sense of it, or do you see it causing an issue in the future if we bring other workflows into the project instead of the swipe classifier?

Reply:
I imagine this will be fine if we start accepting other workflows. `classifier.annotations` is normally in the following format: `{ t0: 1 }` or something like that. So if we, for example, allow for more answers then it will look like `{t0: 3}`. and it will still work fine.